src/Model/HasAttributesTrait.php

```diff
@@ -171,6 +187,13 @@ public function setDocumentAttribute(string $key, $value)
             $value = $this->{$this->buildMutatorMethod($key, 'set')}($value);
         }
 
+        if ($caster = CastResolver::resolve($this->casts[$key] ?? null)) {
+            $value = $caster->set($value);
+            $this->attributes[$key] = $value;
+
+            return;
```

Comment:
Realizar esse ajuste ou repetir tratamento de nulos da linha 197.
```suggestion
            $value = $caster->set($value);
```